medmodels/treatment_effect/builder.py

```diff
@@ -93,12 +93,13 @@ def with_patients_group(self, group: Group) -> TreatmentEffectBuilder:
         return self
 
     def with_time_attribute(
-        self, attribute: MedRecordAttribute
+        self, attribute: Optional[MedRecordAttribute]
```

Comment:
imo this should not be optional. If a user wants to create a treatment effect using the builder without a time attribute, he should just not call this function

Reply:
It might be good to include this in the general docstrings of the Treatment effect builder though. Something along the lines of "by default, it uses static analysis, if you want temporal analysis, you have to define the time attribute using `with_time_attribute`".